tritium-registry/src/test/java/com/palantir/tritium/metrics/registry/MetricNameTest.java

```diff
@@ -36,11 +36,13 @@ public void compareSame() {
                 .putSafeTags("key", "value")
                 .putSafeTags("key1", "value1")
                 .build();
+        MetricName three = RealMetricName.create(two);
 
-        assertThat(one).isEqualTo(two);
-        assertThat(two).isEqualTo(one);
+        assertThat(one).isEqualTo(two).isEqualTo(three);
+        assertThat(two).isEqualTo(one).isEqualTo(three);
+        assertThat(three).isEqualTo(one).isEqualTo(two);
 
-        assertThat(one.toString()).isEqualTo(two.toString());
+        assertThat(one.toString()).isEqualTo(two.toString()).isEqualTo(three.toString());
```

Comment:
minor nit: do we want to verify the actual contents of the `toString` (i.e. there's an unlikely alternate universe where `one == two == three` if we had a flyweight cache behind builder & RealMetricName)?

```suggestion
        assertThat(one.toString())
                .isEqualTo(two.toString())
                .isEqualTo(three.toString())
                .isEqualTo("MetricName{safeName=test, safeTags={key=value, key1=value1, key2=value2}}");
```

Reply:
good call